frontend/lib/resolution/Resolver.h

```diff
@@ -475,37 +470,103 @@ struct Resolver {
   // control flow, and we want to make sure to always keep callNodeStack in sync.
   void handleCallExpr(const uast::Call* call);
 
-  // handle the result of one of the functions to resolve a call. Handles:
-  //  * r.setMostSpecific
-  //  * r.setPoiScope
-  //  * r.setType
-  //  * poiInfo.accumulate
-  //
-  // Does not handle:
-  //
-  //  * issueErrorForFailedCallResolution if there was an error
-  //
-  // Instead, returns 'true' if an error needs to be issued.
-  bool handleResolvedCallWithoutError(ResolvedExpression& r,
-                                      const uast::AstNode* astForErr,
-                                      const CallResolutionResult& c,
-                                      optional<ActionAndId> associatedActionAndId = {});
-  // Same as handleResolvedCallWithoutError, except actually issues the error.
-  void handleResolvedCall(ResolvedExpression& r,
-                          const uast::AstNode* astForErr,
-                          const CallInfo& ci,
-                          const CallResolutionResult& c,
-                          optional<ActionAndId> associatedActionAndId = {});
-  // like handleResolvedCall, but prints the candidates that were rejected
-  // by the error in detail.
-  void handleResolvedCallPrintCandidates(ResolvedExpression& r,
-                                         const uast::Call* call,
-                                         const CallInfo& ci,
-                                         const CallScopeInfo& inScopes,
-                                         const types::QualifiedType& receiverType,
-                                         const CallResolutionResult& c,
-                                         std::vector<const uast::AstNode*>& actualAsts,
-                                         optional<ActionAndId> associatedActionAndId = {});
+  // wraps a CallResolutionResult with additional contextual information
+  // to re-run it if needed and emit error messages. The information
+  // is largely present at the call site for resolve*Call, but
+  // storing it here makes it easy to pass around.
+  struct ResolvedCallResult {
```

Comment:
Feels like it could be confusing to have this name along with ``CallResolutionResult``. Is there another name that could distinguish it better from the preexisting type?

Reply:
I thought so too, but I'm at loss for a better name.

Reply:
Maybe ``CallResultWrapper``? Anyways, I don't want to hold the PR up over this. Up to you 👍